demos/object_detection_demo_ssd_async/main.cpp

```diff
@@ -229,100 +281,93 @@ int main(int argc, char *argv[]) {
                 data[1] = static_cast<float>(netInputWidth);  // width
                 data[2] = 1;
             };
-            setImgInfoBlob(async_infer_request_curr);
-            setImgInfoBlob(async_infer_request_next);
+
+            for (unsigned infReqId = 0; infReqId < FLAGS_nireq; ++infReqId) {
+                InferRequest::Ptr requestPtr = userSpecifiedInferRequests[infReqId];
+                setImgInfoBlob(requestPtr);
+            }
+
+            InferRequest::Ptr requestPtr = minLatencyInferRequest;
+            setImgInfoBlob(requestPtr);
         }
         // -----------------------------------------------------------------------------------------------------
 
-        // --------------------------- 6. Do inference ---------------------------------------------------------
-        slog::info << "Start inference " << slog::endl;
-
-        bool isLastFrame = false;
-        bool isAsyncMode = false;  // execution is always started using SYNC mode
-        bool isModeChanged = false;  // set to TRUE when execution mode is changed (SYNC<->ASYNC)
+        // --------------------------- 6. Init variables -------------------------------------------------------
+        typedef std::chrono::duration<double, std::chrono::milliseconds::period> ms;
+        typedef std::chrono::duration<double, std::chrono::seconds::period> sec;
+        
+        struct RequestResult {
+            cv::Mat frame;
+            const float* output;
+            std::chrono::high_resolution_clock::time_point startTime;
+            bool isSameMode;
+        };
+
+        struct ModeInfo {
+           int framesCount = 0;
+           double latencySum = 0;
```

Comment:
You should use there the original type `std::chrono::steady_clock::duration`. Extract printable value only when you need to report it. With that approach you won't accumulate errors produced by double representation.